server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1612,6 +1613,12 @@ public Ternary<Pair<List<? extends Host>, Integer>, List<? extends Host>, Map<Ho
             logger.debug("Hosts having capacity and suitable for migration: {}", suitableHosts);
         }
 
+        // Only list hosts of the same architecture as the source Host in a multi-arch zone
+        List<CPU.CPUArch> clusterArchs = ApiDBUtils.listZoneClustersArchs(vm.getDataCenterId());
+        if (CollectionUtils.isNotEmpty(clusterArchs) && clusterArchs.size() > 1) {
+            suitableHosts = suitableHosts.stream().filter(h -> h.getArch() == srcHost.getArch()).collect(Collectors.toList());
+        }
```

Comment:
can we do all this only if there are any suitable hosts found?

Reply:
done, it shouldn't have any issues even if it were empty, it would still return an empty list. But I have addressed it.